### ISSUE TITLE
Lights are penetration on walls

### ISSUE THREAD

**Issue by kruglinski** [godotengine/godot#26838]
**Godot version:**

Godot_v3.1-rc1_win64

**OS/device including version:**

OS: Microsoft Windows [版本 10.0.17134.590]
GPU: NVIDIA GeForce GTX 960
Driver: 25.21.14.1917

**Issue description:**

I don't known if this is a bug or not, just seem weird!

![image](https://user-images.githubusercontent.com/2869425/54073145-bcc8b380-42be-11e9-9212-b547647d0595.png)
![image](https://user-images.githubusercontent.com/2869425/54073148-c0f4d100-42be-11e9-83ae-273fd2475236.png)
![image](https://user-images.githubusercontent.com/2869425/54073181-06b19980-42bf-11e9-92b4-2d5ae6c1bfda.png)
![image](https://user-images.githubusercontent.com/2869425/54073185-09ac8a00-42bf-11e9-84fa-93c80bfcf371.png)
![image](https://user-images.githubusercontent.com/2869425/54073188-0d401100-42bf-11e9-81f8-075d8a2c7e7f.png)
![image](https://user-images.githubusercontent.com/2869425/54073189-10d39800-42bf-11e9-879f-3bf92b69703e.png)

**Steps to reproduce:**
Open the godot-sponza-1.0.0 demo project in Godot 3.1 rc1 engine, and look at the second floor

**Minimal reproduction project:**

godot-sponza[](https://github.com/Calinou/godot-sponza)

**Comment by Chaosus** [comment]
Lets tag it a bug, and then sort it out. I pretty sure that one or several duplicated issues are exist about this problem. Also can you outline the places on images where you see a bug ? I see the bug-looking thing only on first image.. you can also try to adjust shadow bias or something which may help you to fix this

**Comment by kruglinski** [comment]
Image 1 and 4 are very obvious, they are all 2nd floor.

On the last image from left to right, second floor light color are different, I think they should not get lights from 1st floor at all, because those lights are under the floor. 

In real world lights can't penetration walls directly

**Comment by Janders1800** [comment]
Happens to me as well, tweaked normal bias and bias, even rebaked the GIProbe.
![Captura de pantalla de 2019-03-09 16-58-24](https://user-images.githubusercontent.com/27329423/54073919-e9fc6e00-428c-11e9-9781-03c2d02a9adb.png)

**Comment by CptPotato** [comment]
When using GI probes *some* leaking is normal, especially when walls are thinner than the cell-size of the GI probe.

**Comment by kruglinski** [comment]
The scene have outer walls, it can show more leaking lights based on distance. far view leaking more lights

![image](https://user-images.githubusercontent.com/2869425/54074722-ee4a7a80-42d0-11e9-92a8-e6bfde13bccf.png)
![image](https://user-images.githubusercontent.com/2869425/54074725-f3a7c500-42d0-11e9-99e8-aff324126263.png)
![image](https://user-images.githubusercontent.com/2869425/54074728-f73b4c00-42d0-11e9-9ed3-5a73306b355d.png)
![image](https://user-images.githubusercontent.com/2869425/54074730-fb676980-42d0-11e9-8147-aeab9537ce3a.png)
![image](https://user-images.githubusercontent.com/2869425/54074733-fefaf080-42d0-11e9-8e36-cf0a6da53fb8.png)
![image](https://user-images.githubusercontent.com/2869425/54074735-028e7780-42d1-11e9-832d-dfdc78b512ea.png)
![image](https://user-images.githubusercontent.com/2869425/54074737-07ebc200-42d1-11e9-818c-459571b8d9da.png)
![image](https://user-images.githubusercontent.com/2869425/54074788-aed05e00-42d1-11e9-81b1-4ccf6c4a303a.png)
![image](https://user-images.githubusercontent.com/2869425/54074789-b2fc7b80-42d1-11e9-9ce4-e8458c996c28.png)
![image](https://user-images.githubusercontent.com/2869425/54074790-b7289900-42d1-11e9-9573-3b5ba7cabf20.png)
![image](https://user-images.githubusercontent.com/2869425/54074792-babc2000-42d1-11e9-98d1-37e01529380b.png)
![image](https://user-images.githubusercontent.com/2869425/54074793-bee83d80-42d1-11e9-875a-62ba91f164bf.png)

**Comment by kruglinski** [comment]
With all directions blocked by walls, outer scene should be very dark, even full black

**Comment by Chaosus** [comment]
I think its a duplicate of #12024, what do you think @akien-mga ?

**Comment by groud** [comment]
IIRC, it is expected to have light going through walls if they are too thin with GIprobes. That's a known drawback of the technique.

**Comment by Megalomaniak** [comment]
Might take contact shadows to mask the GI Probe as a possible fix, maybe?

**Comment by Calinou** [comment]
@Megalomaniak Contact shadows often look broken and cause slowdowns when you approach them, so I'd avoid using them entirely. Moreover, they wouldn't actually hide GI bleeding.
